server/utils/filesystem/ssh.ts

```diff
@@ -3,57 +3,92 @@ import * as path from "node:path";
 import { SSH_PATH } from "@/server/constants";
 import { spawnAsync } from "../process/spawnAsync";
 
-export const generateSSHKey = async (appName: string) => {
+const readSSHKey = async (id: string) => {
+	try {
+		if (!fs.existsSync(SSH_PATH)) {
+			fs.mkdirSync(SSH_PATH, { recursive: true });
+		}
+
+		return {
+			privateKey: fs.readFileSync(path.join(SSH_PATH, `${id}_rsa`), {
+				encoding: "utf-8",
+			}),
+			publicKey: fs.readFileSync(path.join(SSH_PATH, `${id}_rsa.pub`), {
+				encoding: "utf-8",
+			}),
+		};
+	} catch (error) {
+		throw error;
+	}
+};
+
+export const saveSSHKey = async (
+	id: string,
+	publicKey: string,
+	privateKey: string,
+) => {
+	const applicationDirectory = SSH_PATH;
+
+	const privateKeyPath = path.join(applicationDirectory, `${id}_rsa`);
+	const publicKeyPath = path.join(applicationDirectory, `${id}_rsa.pub`);
+
+	const privateKeyStream = fs.createWriteStream(privateKeyPath, {
+		mode: 0o400,
+	});
+	privateKeyStream.write(privateKey);
+	privateKeyStream.end();
+
+	const publicKeyStream = fs.createWriteStream(publicKeyPath, { mode: 0o400 });
```

Comment:
I think is not necessary to have too restrictive permissions on a public key, we can use the standard permission which is 644

Reply:
Sure! I pushed a new commit with `fs.writeFileSync`, which sets 644 by default.